### PR TITLE
Improve some log

### PR DESCRIPTION
**Pull request by binlijin**
improve for debug

### CODE REVIEW

File: server/src/main/java/io/druid/server/QueryResource.java [hunk 1]
```diff
@@ -107,9 +107,9 @@ public QueryResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Response getServer(@PathParam("id") String queryId)
   {
+    log.info("Get cancel query for queryId[%s]", queryId);
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/QueryResource.java]: Debug level please.

Also suggest "Received cancel request for query [%s]"  or similar

File: server/src/main/java/io/druid/server/QueryResource.java [hunk 2]
```diff
@@ -151,6 +152,7 @@ public Response doPost(
         );
       }
 
+      Thread.currentThread().setName("ProcessQuery-"+queryId);
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/QueryResource.java]: At this point the code should be in a jetty thread, not a processing thread. Suggest `"qtp-query-" + queryId`

Comment (drcrallen) [inline server/src/main/java/io/druid/server/QueryResource.java]: Also I'm not sure this method blocks. I thought it blocked in the response's StreamingOutput (but haven't profiled this code area in a very long time)

File: server/src/main/java/io/druid/server/QueryResource.java (hunk 1 above)
Comment (binlijin) [inline server/src/main/java/io/druid/server/QueryResource.java]: ok, done

File: server/src/main/java/io/druid/server/QueryResource.java (hunk 2 above)
Comment (binlijin) [inline server/src/main/java/io/druid/server/QueryResource.java]: Ok, done.
Also I think this do not affect the performance much.

**Comment by fjy** [comment]
👍  can you squash?

File: server/src/main/java/io/druid/server/QueryResource.java [hunk 3]
```diff
@@ -151,6 +154,7 @@ public Response doPost(
         );
       }
 
+      Thread.currentThread().setName("qtp-query-" + queryId);
```
Comment (xvrl) [inline server/src/main/java/io/druid/server/QueryResource.java]: what's the purpose of setting the thread name, is it just for logging or for some other debugging purposes?

**Comment by drcrallen** [comment]
:+1: after @xvrl 's questions are answered

File: server/src/main/java/io/druid/server/QueryResource.java (hunk 3 above)
Comment (xvrl) [inline server/src/main/java/io/druid/server/QueryResource.java]: a more elegant approach might be to use log4j's diagnostic contexts http://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/NDC.html and http://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/MDC.html

Comment (drcrallen) [inline server/src/main/java/io/druid/server/QueryResource.java]: SLF4J's MDC should be compatible with the Log4j one. If MDC is elected here, I suggest using that one.

Comment (binlijin) [inline server/src/main/java/io/druid/server/QueryResource.java]: @xvrl @drcrallen  this is for debugging purposes, not for logging.

**Comment by xvrl** [comment]
@binlijin my concern changing the thread name mid-way through execution might actually make things more confusing to debug for someone that's not familiar with the code.

**Comment by binlijin** [comment]
@xvrl, ok, i will revert this and keep the others.

**Comment by drcrallen** [comment]
@binlijin it was discussed in the dev-sync that it could also be appropriate to take the approach in `io.druid.query.spec.SpecificSegmentQueryRunner#doNamed`

Check out the video form the dev-sync today for the whole conversation: https://www.youtube.com/watch?v=H973WfFz4GE

I think @gianm was going to make a comment but I completely forget what it was supposed to be.

**Comment by drcrallen** [comment]
hopefully @xvrl can elaborate on if he feels it should be either completely removed, or made more like the specific segment query runner.

**Comment by binlijin** [comment]
@drcrallen, thanks for the information, currently i can not open the video from my network in china...

**Comment by gianm** [comment]
@drcrallen @binlijin what I had said in the dev sync was that I think it does make sense to rename the thread. it tells you what query is running in that thread, which is helpful when looking at thread dumps or jconsole.

IMO the best way to do this is something like what SpecificSegmentQueryRunner does: rename the thread inside a try/finally where the "finally" changes the name back to the original name. And for the thread name, IMO a nice one is:

```
String.format("%s[%s_%s_%s]", currentThreadName, queryType, dataSource, queryId)
```

So that would give thread names like,

```
qtp1706655559-42[topN_mydatasource_queryUUID]
```

**Comment by binlijin** [comment]
@gianm, it is a very good idea.

**Comment by drcrallen** [comment]
@binlijin before you close and open a PR to get travis to pass, can you log what the transient error is?

**Comment by binlijin** [comment]
@drcrallen, see  https://travis-ci.org/druid-io/druid/builds/122679173 and https://travis-ci.org/druid-io/druid/jobs/122679174
The error is : 
```
Tests run: 23, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 139.41 sec <<< FAILURE! - in io.druid.indexing.kafka.KafkaIndexTaskTest
testRunAfterDataInserted[buildV9Directly = false](io.druid.indexing.kafka.KafkaIndexTaskTest)  Time elapsed: 7.866 sec  <<< ERROR!
org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 6000
	at org.I0Itec.zkclient.ZkClient.connect(ZkClient.java:1223)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:155)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:129)
	at kafka.utils.ZkUtils$.createZkClientAndConnection(ZkUtils.scala:89)
	at kafka.utils.ZkUtils$.apply(ZkUtils.scala:71)
	at kafka.server.KafkaServer.initZk(KafkaServer.scala:278)
	at kafka.server.KafkaServer.startup(KafkaServer.scala:168)
	at io.druid.indexing.kafka.test.TestBroker.start(TestBroker.java:69)
	at io.druid.indexing.kafka.KafkaIndexTaskTest.setUp(KafkaIndexTaskTest.java:245)
testRunAfterDataInserted[buildV9Directly = false](io.druid.indexing.kafka.KafkaIndexTaskTest)  Time elapsed: 7.866 sec  <<< ERROR!
java.lang.NullPointerException: null
	at io.druid.indexing.kafka.KafkaIndexTaskTest.tearDown(KafkaIndexTaskTest.java:271)
Results :
Tests in error: 
io.druid.indexing.kafka.KafkaIndexTaskTest.testRunAfterDataInserted[buildV9Directly = false](io.druid.indexing.kafka.KafkaIndexTaskTest)
  Run 1: KafkaIndexTaskTest.setUp:245 » ZkTimeout Unable to connect to zookeeper server...
  Run 2: KafkaIndexTaskTest.tearDown:271 NullPointer
```

**Comment by binlijin** [comment]
@xvrl , what about now?

**Comment by gianm** [comment]
👍

**Comment by fjy** [comment]
👍

**Pull request merged**